content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/system-extensions.md

```diff
@@ -0,0 +1,48 @@
+---
+linktitle: Extending the Terrain System
+title: Extending the Terrain System
+description: Information on how to extend the terrain system to accomplish various goals.
+weight: 400
+---
+The terrain system is designed to be highly decoupled and easily extensible at a number of different layers.
+
+## Gradient Components
+
+Gradient components are the underlying data source for height data and surface weight data. New gradient component types can be created as-needed to provide data from different sources, such as streaming satellite data, or to modify existing gradient data, such as adding complex erosion filters.
+
+New gradient components need to support the following at a minimum:
+
+* Implement the `GradientRequestBus` to generically return a floating-point output value for an input world position.
+* Declare that it implements the `GradientService` in `GetProvidedServices`.
+* Support thread safety. Ideally, this will mean using a `shared_mutex` with a `shared_lock` in the query APIs and a `unique_lock` for any places that change the underlying data used in the query API methods. This will let multiple queries run in parallel while safely blocking on data changes.
+* Call `DependencyNotificationBus::Events::OnCompositionChanged` or `DependencyNotificationBus::Events::OnCompositionRegionChanged` whenever any data within the gradient changes. This notifies everything "above" the gradient that it will need to refresh itself.
+
+Ideally, new gradient components would also support the following:
```

Comment:
I suggest using an alternative to "ideally" to circumvent explaining why this is ideal, which readers might wonder. "should" is the more appropriate form here. 
```suggestion
In addition, new gradient components should also support the following:
```